octopoes/octopoes/core/service.py

```diff
@@ -45,6 +46,15 @@
 
 logger = getLogger(__name__)
 settings = Settings()
+BIT_CACHE: dict[int, list[OOI]] = {}
+
+
+def bit_cache_key(*args) -> int:
```

Comment:
I'd rather have a discrete Hash method be implemented on the OOI base class. This would allow us to also subclass it for specific OOI-types, and ignore some fields (albeit introducing issues with bits that act on those fields in that case). Pickling the whole object is a security issue (a bit theoretical at the moent though), and probably covers too many of the objects members.

Reply:
Removed the pickle

Reply:
We could base the bit cache key on:
- `bit_definition.id`
- `source.model_dump_json()`
- some composition of `model_dump_json` for parameters
- `json.dumps(config, sort_keys=True)`

This does not work for Mock objects though.

Reply:
Addressed in a9ff441948996c1e6d66a3464c5d5e2f95088da9.